### DISCUSSION TITLE
How to set Padding to UnformGridLayout ?

### DISCUSSION THREAD

**Discussion by Ready4Next** [AvaloniaUI/Avalonia#9746] [Q&A] [answered]
Hi,

I have an ItemsRepeater using an UniformGridLayout to display all the items in a list.

```xml
<ItemsRepeater Items="{Binding AllGames, Mode=OneWay}">
				<ItemsRepeater.Layout>
					<UniformGridLayout Orientation="Horizontal" />
				</ItemsRepeater.Layout>
				<ItemsRepeater.ItemTemplate>
					<DataTemplate>
...
</DataTemplate>
				</ItemsRepeater.ItemTemplate>					
			</ItemsRepeater>	
```

However, each element in the list have a scale animation on MouseHover, which results in overflows at the top and left of the item.

![image](https://user-images.githubusercontent.com/1833279/208624036-0a2bc50b-3754-4320-8fe9-7cca866d1c7f.png)

I thought of adding left and top padding to the ItemsRepeater, which uses UnformGridLayout, but there's no padding and LayoutOrigin is internal. Or Margin to UniformGridLayout. 

Do you have any ideas about how I can achieve that?

Thanks a lot,

**Comment by maxkatz6** [comment, marked as answer]
> However, each element in the list have a scale animation on MouseHover, which results in overflows at the top and left of the item.

Feels like you need to set ClipToBounds=False on the elements. You should check with DevTools and visual tree to find out which elements might be clipped.

> I thought of adding left and top padding to the ItemsRepeater, which uses UnformGridLayout, but there's no padding and LayoutOrigin is internal. Or Margin to UniformGridLayout.

UniformGridLayout is not a control by itself. You need to set Padding either on whole ItemsRepeater or on each item.

**Reply by Ready4Next** [reply]
Hi,

I will try ClipToBounds on the main element that contains the ItemDetails, which is a Border (and it's the Border that is scaled). I don't know how it works, but I'm glad to learn a new thing. Meanwhile, I have done some ugly hacks like that:

Setting row/col spacing with spacing around the item:

```xml
<UniformGridLayout Orientation="Horizontal" MinColumnSpacing="2.5" MinRowSpacing="10" ItemsJustification="SpaceAround" />
```

And in the ItemTemplate which contains the border:

```xml
<ItemsRepeater.ItemTemplate>
					<DataTemplate>
												
						<Border HorizontalAlignment="Center" 
								MaxWidth="150"
								Margin="5 20 0 0"
								VerticalAlignment="Center">
[...]
```

Which I don't like because it adds extra space between the elements. See video below:

https://user-images.githubusercontent.com/1833279/208745704-fbb6ebfc-923a-4f2c-b499-13b87f95f39d.mp4

That's a simple testing project, the main purpose is to learn with something that will keep me motivated ;-)

I will update this answer after playing around with ClipToBounds.

Thanks for your kind answer again.

**Reply by Ready4Next** [reply]
Again, you were right, no more hacky margin things, and I was wrong with modifying within the ItemTemplate. Here is with ClipToBounds => True (I intentionally exaggerated the zoom for the video):

https://user-images.githubusercontent.com/1833279/208781030-12de4c25-33ce-40f2-badb-b45cf386f2f3.mp4

And now => False:

https://user-images.githubusercontent.com/1833279/208781075-fd425485-5698-4b87-8139-23b5e10cd2c5.mp4

Code:

```xml
<ItemsRepeater Items="{Binding AllGames, Mode=OneWay}" Margin="20 30 20 0" ClipToBounds="False">
				<ItemsRepeater.Layout>
					<UniformGridLayout Orientation="Horizontal" 
									   MinColumnSpacing="20" MinRowSpacing="30" 
									   ItemsJustification="SpaceAround" />
				</ItemsRepeater.Layout>
				<ItemsRepeater.ItemTemplate>
					<DataTemplate>
												
						<Border HorizontalAlignment="Center" 
								MaxWidth="150"								
								VerticalAlignment="Center">
[...]
```

Thanks a lot !
